Ical.Net/DataTypes/CalDateTime.cs

```diff
@@ -470,20 +427,40 @@ public IDateTime ToTimeZone(string otherTzId)
     }
 
     /// <inheritdoc cref="DateTime.Add"/>
-    /// <remarks>
-    /// This will also add a <see cref="IDateTime.Time"/> part that did not exist before the operation,
-    /// if the hours are not a multiple of 24.
-    /// </remarks>
-    public IDateTime Add(TimeSpan ts) => this + ts;
+    /// <exception cref="InvalidOperationException">
+    /// Thrown when attempting to add a time span to a date-only instance, 
+    /// and the time span is not a multiple of full days.
+    /// </exception>
+    public IDateTime Add(TimeSpan ts)
+    {
+        if (!HasTime && (ts.Ticks % TimeSpan.TicksPerDay) != 0)
+        {
+            throw new InvalidOperationException("This instance represents a 'date-only' value. Only multiples of full days can be added to a 'date-only' instance.");
+        }
+
+        // Ensure to handle DST transitions correctly when timezones are involved.
+        var newDateTime = TzId is null ? Value.Add(ts) : AsUtc.Add(ts);
+
+        var copy = Copy<CalDateTime>();
+        copy._dateOnly = DateOnly.FromDateTime(newDateTime);
+        copy._timeOnly = HasTime ? TruncateTimeToSeconds(newDateTime) : null;
+        copy._tzId = TzId is null ? null : UtcTzId;
+
+        return TzId is null ? copy : copy.ToTimeZone(TzId);
+    }
 
     /// <summary>Returns a new <see cref="TimeSpan" /> from subtracting the specified <see cref="IDateTime"/> from to the value of this instance.</summary>
     /// <param name="dt"></param>
-    public TimeSpan Subtract(IDateTime dt) => (AsUtc - dt.AsUtc)!;
+    public TimeSpan Subtract(IDateTime dt) => AsUtc - dt.AsUtc;
 
     /// <summary>Returns a new <see cref="IDateTime"/> by subtracting the specified <see cref="TimeSpan" /> from the value of this instance.</summary>
     /// <param name="ts">An interval.</param>
     /// <returns>An object whose value is the difference of the date and time represented by this instance and the time interval represented by <paramref name="ts" />.</returns>
-    public IDateTime Subtract(TimeSpan ts) => this - ts;
+    /// <exception cref="InvalidOperationException">
+    /// Thrown when attempting to subtracting a time span from a date-only instance, 
+    /// and the time span is not a multiple of full days.
+    /// </exception>
+    public IDateTime Subtract(TimeSpan ts) => Add(-ts);
```

Comment:
Do we need a Subtract at all? It doesn't exist in DateTime either.

Reply:
tracked with #677